bob-backend/src/pearl/group.rs

```diff
@@ -170,54 +172,37 @@ impl Group {
     // create pearl for current write
     async fn create_write_pearl(
         &self,
-        ts: u64,
+        data: &BobData,
         timestamp_config: StartTimestampConfig,
     ) -> Result<(ChildId, Holder), Error> {
-        let mut indexes = self.created_holder_indexes.write().await;
-        let created_holder_index = indexes.get(&ts).copied();
-        let index = if let Some(exisiting_index) = created_holder_index {
-            exisiting_index
+        let holders = self.holders.upgradable_read().await;
+        
+        let created_holder_index = Self::find_actual_holder(&holders, data).await;
+        Ok(if let Ok(index_and_holder) = created_holder_index {
+            index_and_holder
         } else {
-            let new_index = self
+            info!("creating pearl for timestamp {}", data.meta().timestamp());
+            let pearl = self.create_pearl_by_timestamp(data.meta().timestamp(), &timestamp_config);
+            self
                 .settings
                 .config()
                 .try_multiple_times_async(
-                    || self.try_create_write_pearl(ts, &timestamp_config),
+                    || pearl.prepare_storage(),
```

Comment:
You should preserve old behavior: run first `create_pearl_by_timestamp` and the `prepare_storage`. 
It is very likely that in the case of a error, the `holder` needs to be recreated.